lib/Sema/CSSyntacticElement.cpp

```diff
@@ -2208,29 +2383,15 @@ bool ConstraintSystem::applySolutionToBody(Solution &solution,
   // transformations.
   llvm::SaveAndRestore<DeclContext *> savedDC(currentDC, fn.getAsDeclContext());
 
-  Type resultTy;
-
-  if (auto transform = solution.getAppliedBuilderTransform(fn)) {
-    resultTy = solution.simplifyType(transform->bodyResultType);
-  } else if (auto *closure =
-                 getAsExpr<ClosureExpr>(fn.getAbstractClosureExpr())) {
-    resultTy =
-        solution.getResolvedType(closure)->castTo<FunctionType>()->getResult();
-  } else {
-    resultTy = fn.getBodyResultType();
-  }
-
   SyntacticElementSolutionApplication application(
-      solution, SyntacticElementContext::forFunctionRef(fn), resultTy,
-      rewriteTarget);
+      solution, SyntacticElementContext::forFunctionRef(fn), rewriteTarget);
 
-  auto body = application.apply();
+  auto *body = application.apply();
 
   if (!body || application.hadError)
     return true;
 
-  fn.setTypecheckedBody(castToStmt<BraceStmt>(body),
-                        fn.hasSingleExpressionBody());
+  fn.setTypecheckedBody(cast<BraceStmt>(body), fn.hasSingleExpressionBody());
```

Comment:
Hm, I think we have a bug here unrelated to your changes - if function/accessor/closure has been result builder transformed it would no longer be `single-expression` so we need to expand the check.

Reply:
Heh, good catch! I can file a bug if you want

Reply:
Let's do that so it doesn't get lost. I'll open a PR to fix that once your changes land.

Reply:
https://github.com/apple/swift/issues/63264